streaming/src/main/scala/org/apache/gearpump/streaming/appmaster/TaskSchedulerImpl.scala

```diff
@@ -17,26 +17,51 @@
  */
 package org.apache.gearpump.streaming
 
+import com.typesafe.config.Config
 import org.apache.gearpump.cluster.scheduler.{Relaxation, Resource, ResourceRequest}
 import org.apache.gearpump.streaming.TaskLocator.{Locality, NonLocality, WorkerLocality}
-import org.apache.gearpump.streaming.TaskSet.TaskLaunchData
+import org.apache.gearpump.streaming.TaskSchedulerImpl.TaskLaunchData
 import org.apache.gearpump.streaming.task.TaskId
 import org.apache.gearpump.util.LogUtil
 import org.slf4j.Logger
 
 import scala.collection.mutable
 
-class TaskSet(appId : Int,  dag : DAG) {
+trait TaskScheduler {
+  def dagUpdated(dag: DAG): Array[ResourceRequest]
+  def resourceAllocated(workerId : Int, executorId: Int) : Option[TaskLaunchData]
+  def resourceFailed(executorId: Int) : Array[ResourceRequest]
+}
+
+class TaskSchedulerImpl(appId : Int, config: Config)  extends TaskScheduler {
   private val LOG: Logger = LogUtil.getLogger(getClass, app = appId)
 
   private var taskMap = Map.empty[TaskId, TaskLaunchData]
   private var taskQueues = Map.empty[Locality, mutable.Queue[TaskLaunchData]]
-  private val taskLocator = new TaskLocator
-  var totalTaskCount = 0
+  private val taskLocator = new TaskLocator(config)
 
-  init(dag)
+  private var executorToTaskIds = Map.empty[Int, Set[TaskId]]
 
-  def fetchResourceRequests(fromOneWorker: Boolean = false): Array[ResourceRequest] ={
+  def dagUpdated(dag: DAG): Array[ResourceRequest] ={
```

Comment:
the return value is quite "surprising" and has nothing to do with "dagUpdated". Also, `fetchResourceRequests(false)` seems not to be related to previous codes. "dagUpdated" is like a result but doesn't reflect what's done in the method.